### DISCUSSION TITLE
Can't start Python console on remote?

### DISCUSSION THREAD

**Discussion by astrowonk** [posit-dev/positron#5296] [Q&A] [answered]
I was about to open an issue, but I thought I'd post here first in case this is a known problem.

I connect successfully with SSH Remote to a remote Linux (Red hat) machine. I can view files, edit code, etc. It discovers and I can select an interpreter (so this seems different than #4715) But If I try to open the REPL console, it fails with:

 Cannot read properties of undefined (reading 'createSession')

and in the Dev tools log:

```
 INFO Starting session for language runtime 4f7c2468ba8380db76fbb2a9abd46cc9 (language: Python name: Python 3.11.9 (Conda: base) version: 3.11.9) (Source: User-requested new session from console action bar after session python-2d85279e exited.)
workbench.desktop.main.js:130713 DEBUG [Ext host 0] Runtime manager for 'Python 3.11.9 (Conda: base)': false
workbench.desktop.main.js:130713 DEBUG [Ext host 1] Runtime manager for 'Python 3.11.9 (Conda: base)': true
workbench.desktop.main.js:130743   ERR Starting language runtime failed. Reason: Cannot read properties of undefined (reading 'createSession')
workbench.desktop.main.js:130743   ERR Cannot read properties of undefined (reading 'createSession')
```

Any troubleshooting steps or is this a bug? Thanks!

**Comment by jmcphers** [comment]
This sounds like it could be a bug, but this works for most folks so there's likely something unsupported in your environment.

- Do you have a "Jupyter Adapter" output channel (in the Output tab adjacent to the Console)? If so, what does it show?
- What version of Positron are you using?
- What version of Red Hat is running on the remote machine?

**Reply by astrowonk** [reply]
Hmm, no I don't have a "Jupyter Adapter". I'm running the latest Mac version I think , I dowloaded it today.

```
Positron Version: 2024.11.0 (Universal) build 140
Code - OSS Version: 1.93.0
Commit: e0d844b031f95acbf89f234a2cce2af9b6721f6c
Date: 2024-10-31T14:24:19.540Z
Electron: 30.4.0
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0
```

The remote is ` Red Hat Enterprise Linux 8.10 (Ootpa)`

It may be a symptom of a bigger problem because I tried making a new jupyter notebook inside of Positron and it opens, and I pick the intepreter, but nothing actually works. Everything (even a print statement) gets a red X and if I try to restart the notebook kernel it fails with a timeout.

```
2024-11-07 15:11:13.728 [error] Error: Timeout waiting for runtime to restart
	at Timeout._onTimeout (/home/236223/.positron-server/bin/e0d844b031f95acbf89f234a2cce2af9b6721f6c/extensions/positron-notebook-controllers/dist/extension.js:1:17400)
	at listOnTimeout (node:internal/timers:573:17)
	at process.processTimers (node:internal/timers:514:7)
	
```

**Reply by jmcphers** [reply, marked as answer]
It's a known issue that Positron doesn't work on Red Hat 8, unfortunately. It uses an old glibc and openssl version that we'd need to make a special build for. See here for a bunch more info/notes: https://github.com/posit-dev/positron/issues/3854

**Reply by astrowonk** [reply]
Yikes, yeah I see the GLIB 2.33 error now in the developer console. And apparently, unlike problems with `libstdc++` which we have worked around by linking to one that comes with miniconda, tinkering with libc.so can break everything. (I successfully compiled 2.33, but every post I find on the internet says I'll break everything if I tried to just symlink it in as a replacement.)

Shout out @isabelizimm's PyData talk for getting me to try the remote ssh feature! I'll have to use Positron locally for now at home.
